src/lib/Server/Controller/JWT.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformRest\Server\Controller;
+
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\Core\Base\Exceptions\UnauthorizedException;
+use eZ\Publish\Core\MVC\Symfony\Security\User;
+use EzSystems\EzPlatformRest\Message;
+use EzSystems\EzPlatformRest\Server\Controller as RestController;
+use EzSystems\EzPlatformRest\Server\Values;
+use Lexik\Bundle\JWTAuthenticationBundle\Services\JWTTokenManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+
+class JWT extends RestController
+{
+    /** @var \eZ\Publish\API\Repository\UserService */
+    private $userService;
+
+    /** @var \Lexik\Bundle\JWTAuthenticationBundle\Services\JWTTokenManagerInterface */
+    private $tokenManager;
+
+    public function __construct(
+        UserService $userService,
+        JWTTokenManagerInterface $tokenManager
+    ) {
+        $this->userService = $userService;
+        $this->tokenManager = $tokenManager;
+    }
+
+    public function createToken(Request $request): Values\JWTToken
+    {
+        /** @var \EzSystems\EzPlatformRest\Server\Values\JWTTokenInput $jwtTokenInput */
+        $jwtTokenInput = $this->inputDispatcher->parse(
+            new Message(
+                ['Content-Type' => $request->headers->get('Content-Type')],
+                $request->getContent()
+            )
+        );
+
+        try {
+            $user = $this->userService->loadUserByLogin($jwtTokenInput->username);
```

Comment:
Do you think we could refactor that part, up to returning the token, so that it can also be used from a GraphQL implementation ?

A simple service that given a username & password returns a token, or throws clearly documented exceptions.

Reply:
Yeah, why not but then it should definitely be placed in the kernel instead of REST, right

Reply:
Yep, I agree with that. You can see in https://github.com/ezsystems/ezplatform-graphql/pull/86 that I have the exact same code.

Reply:
> > Do you think we could refactor that part, up to returning the token, so that it can also be used from a GraphQL implementation ?
> > 
> > A simple service that given a username & password returns a token and throws clearly documented exceptions.

> Yeah, why not but then it should definitely be placed in the kernel instead of REST, right?

Can this be somewhere in ezplatform-core then, maybe? I don't want to add coupling on JWT to kernel, there are too many couplings there already.

Reply:
I have no problem with it being in core.